src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -34,6 +37,11 @@ internal class MsalServiceExceptionFactory
                 {
                     ex = new MsalUiRequiredException(errorCode, MsalErrorMessage.AadThrottledError, innerException);
                 }
+                else if (oAuth2Response.Claims != null)
```

Comment:
I don't see why the throttled exception can't have Claims. It looks like just the error message is different.